symengine/polynomial.cpp

```diff
@@ -566,25 +555,21 @@ RCP<const UnivariatePolynomial> add_uni_poly(const UnivariatePolynomial &a,
     } else {
         var = a.get_var();
     }
-    for (const auto &it : a.get_dict())
-        dict[it.first] = it.second;
-    for (const auto &it : b.get_dict())
-        dict[it.first] += it.second;
+    map_int_Expr dict;
+    dict = (a.get_dict2() + b.get_dict2()).get_dict();
     return univariate_polynomial(var, std::move(dict));
```

Comment:
After fixing the signature of `univariate_polynomial`, this should be,
```
UnivariateExprPolynomial dict = a.get_dict2();
dict += b.get_dict2();
```
to avoid copying.